src/dbnode/persist/fs/merger.go

```diff
@@ -89,7 +89,7 @@ func (m *merger) Merge(
 	flushPreparer persist.FlushPreparer,
 	nsCtx namespace.Context,
 	onFlush persist.OnFlushSeries,
-) (err error) {
+) (closer persist.DataCloser, err error) {
```

Comment:
nit: Can we avoid using named returns? Looks like they're not being used (no "return" statements) and it's too easy to over shadow accidently.

Maybe best at top of func to just declare `var closer persist.DataCloser`?